erts/emulator/beam/packet_parser.c

```diff
@@ -533,15 +533,37 @@ http_parse_absoluteURI(PacketHttpURI* uri, const char* uri_ptr, int uri_len)
 
     uri->s1_ptr = uri_ptr;
     uri->port = 0; /* undefined */
-    /* host[:port]  */
     if ((p = memchr(uri_ptr, ':', uri_len)) == NULL) {
         uri->s1_len = uri_len;
     }
+    /* ipv6
+     * eg [::1]:4000
+     * eg [FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80
+     */
+    else if (memchr(uri_ptr, '[', uri_len) == uri_ptr) {
+      if ((p = memchr(uri_ptr, ']', uri_len)) != NULL) {
```

Comment:
I don't think this is correct.
If there is a `[` without terminating `]` then `uri->s1_len` will not be set leading to potential disaster.

It should probably be combined into one if-statement.

Reply:
I combined both if statements into a single one in the latest commit.